config_test.go

```diff
@@ -80,6 +109,23 @@ func TestConfig(t *testing.T) {
 			for i := 0; i < 200; i++ {
 				logger.Info("sampling")
 			}
+
+			// Rolling over tick to produce sampling report.
+			if tt.expectSamplingN > 0 {
+				ztest.Sleep(time.Second)
+				logger.Info("sampling")
+				byteContents, err = ioutil.ReadAll(temp)
+				require.NoError(t, err, "Couldn't read log contents from temp file.")
+				logs = string(byteContents)
+				logSlice := strings.Split(logs, "\n")
+				samplingRep := logSlice[len(logSlice)-3]
+				require.Regexpf(t, tt.expectSamplingRe, samplingRep, "Unexpected sampling report output.")
+				re := regexp.MustCompile(`[\d]+`)
```

Comment:
nit: regexp simpler as just `\d+`
nit: prefer to pre-compile regexps at package level, ala
```
var numberPattern = regexp.MustCompile(`\d+`)
```